starknet-replay/src/profiler/replay_statistics.rs

```diff
@@ -163,4 +164,45 @@ impl ReplayStatistics {
         );
         filtered_libfuncs
     }
+
+    /// Serialises `ReplayStatistics` to CSV format.
+    ///
+    /// Libfuncs are reported in ascending order of weight.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Err`] if there is an IO error writing to the buffer.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use std::str;
+    /// # use indoc::indoc;
+    /// # use starknet_replay::ReplayStatistics;
+    /// let mut replay_statistics = ReplayStatistics::default();
+    /// replay_statistics.update(&"u32_to_felt252".to_string(), 759);
+    /// replay_statistics.update(&"const_as_immediate".to_string(), 264);
+    /// let expected_string = indoc! {r#"
+    ///     Function Name,Weight
+    ///     const_as_immediate,264
+    ///     u32_to_felt252,759
+    /// "#};
+    /// let csv_output = replay_statistics.to_csv().unwrap();
+    /// assert_eq!(
+    ///     str::from_utf8(csv_output.as_slice()).unwrap(),
+    ///     expected_string
+    /// );
+    /// ```
+    pub fn to_csv(&self) -> Result<Vec<u8>, std::io::Error> {
```

Comment:
The name here implies that it should output a textual format, but you are outputting bytes. Perhaps `to_csv_bytes` instead?